CRR/ReplicationStatusUpdater.js

```diff
@@ -172,36 +175,54 @@ class ReplicationStatusUpdater {
                 //   codebase easier to maintain and upgrade, as opposed to having multiple branches or versions of
                 //   the code for different schema versions.
                 objMD = new ObjectMD(JSON.parse(mdRes.Body));
-                if (!this._objectShouldBeUpdated(objMD)) {
+                if (!this._objectShouldBeUpdated(objMD, storageClass)) {
                     skip = true;
                     return process.nextTick(next);
                 }
                 // Initialize replication info, if missing
                 // This is particularly important if the object was created before
                 // enabling replication on the bucket.
-                if (!objMD.getReplicationInfo()
-                    || !objMD.getReplicationSiteStatus(storageClass)) {
+                let replicationInfo = objMD.getReplicationInfo();
+                if (!replicationInfo || !replicationInfo.status) {
                     const { Rules, Role } = repConfig;
                     const destination = Rules[0].Destination.Bucket;
```

Comment:
this may not be correct, if there are more than 1 desitination....

Reply:
`crrExistingObjects` doesn't currently support triggering replication for multiple sites at once. So this would work as it will initialize the replication info when triggering the first site and then append the other sites' info to the fields when the other ones are triggered.

Reply:
in cloudserver, this is initialized to `bucketMD.replicationConfig.destination`

Reply:
This is a weird one... So, we currently only support having a single destination bucket for all replication rules of a bucket. When creating the rules/workflows via UI it's even worse, the destination bucket becomes the name of the bucket we are replicating from.
The value of this field is not used in Zenko. When creating a location, Cloudserver initializes a client class for the respective backend (aws/azure/...) that keeps the name of the destination bucket in memory, that's the value we use when replicating and not what's in the replication rule.